src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -118,6 +118,8 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
             contributionTimeoutTaskDetector.detect();
         }
 
+        // TODO : Remove this, the optional can never be empty
+        // This is covered in workerService.canAcceptMoreWorks
```

Comment:
#getWorker should be done before #getInitializedOrRunningTasks to avoid losing time. I would add most safety checks in  previous task stream filters (before this for or directly in the database query).

Reply:
In fact, this call only serve to retrieve worker instance for `worker.isTeeEnabled()` check.
If we want to go further right now, I would prefer to access TEE worker status through a WorkerService call.
It may be best to do it in another PR to avoid mixing 2 subjects. That's why I added TODO on lines 121 and 146.

I agree that I would prefer to have:
`boolean isWorkerTeeEnabled = workerService.isTeeEnabled(walletAddress)`
outside from the loop to only compute the status once